### DISCUSSION TITLE
Nested Serialization By Alias

### DISCUSSION THREAD

**Discussion by AbbasCiatek** [fastapi/fastapi#11095] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get(
    path="/me",
    response_model=UserDTO,
    tags=["Auth"],
    response_model_by_alias=True,
)
def me(auth_user: UserDTO = Depends(AuthorizeUser())) -> UserDTO:
    return auth_user

class UserDTO(AbstractModelDTO):
   ...
    department: Annotated[
        Optional[DepartmentDTO],
        Field(alias="department")
    ]


class DepartmentDTO(AbstractModelDTO):
    ....
    permissions: Annotated[
        str,
        Field(
            alias="Permissions",
        )
    ]
```

### Description

This will return permissions as `permisons`   instead of `Permissions`  , I already have a function that recursively calls model_dump on the nested attributes if an attr has model dump and gets me a dict but with alias but I don't know how to override fastapi serializer. If there is another solution that doesn't have me override the default serializer it will also be great. I can also remove the the response model but that will ruin the openapi schema.

### Operating System

Linux

### Operating System Details

Archlinux

### FastAPI Version

fastapi==0.105.0

### Pydantic Version

2.6.0

### Python Version

Python 3.11.6

**Comment by msehnout** [comment, marked as answer]
What about writing it like this:

```python
class DepartmentDTO(BaseModel):
    permissions: str = Field(serialization_alias="Permissions")


class UserDTO(BaseModel):
    department: DepartmentDTO | None = Field(alias="department")


@app.get("/departments")
async def departments() -> UserDTO:
    return UserDTO(department=DepartmentDTO(permissions="abc"))
```

It results in:
```
$ curl http://localhost:8000/departments
{"department":{"Permissions":"abc"}}
```
